common/environment.go

```diff
@@ -87,8 +88,14 @@ func (EnvironmentVariable) UserDir() EnvironmentVariable {
 
 var EAutoLoginType = AutoLoginType(0)
 
+// Wrapper type for AutoLoginType that handles custom JSON marshaling and unmarshaling
 type AutoLoginType uint8
 
+// Wrapper type for AutoLoginType that handles custom JSON marshaling and unmarshaling
+type AutoLoginTypeJSON struct {
```

Comment:
I think a wrapper type is unnecessary here, any defined type can have functions attached. I *believe* `json.Unmarshal` and `json.Marshal` always check even if the underlying type is a primitive(?), but lmk if I'm wrong.

Reply:
Updated